go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -225,6 +228,56 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 	existingNode, err := state.Node(newNode.ID)
 	isNewNode := err == registry.ErrNoSuchNode
 	isExpiredNode := err == nil && existingNode.IsExpired(uint64(epoch))
+	if !isNewNode && err != nil {
+		// Something went horribly wrong, and we failed to query the node.
+		app.logger.Error("RegisterNode: failed to query node",
+			"err", err,
+			"new_node", newNode,
+			"existing_node", existingNode,
+			"entity", newNode.EntityID,
+		)
+		return registry.ErrInvalidArgument
+	}
+
+	// For each runtime the node registers for, require it to pay a maintenance fee for
+	// each epoch the node is registered in.
+	additionalEpochs := newNode.Expiration - uint64(epoch)
```

Comment:
It is possible to create a new node, that expires in the current epoch (additionalEpochs = 0), without paying gas.  I don't think the node will actually do anything, but that doesn't seem great.

Reply:
I changed the expiry check above to explicitly disallow registering nodes which expire in the current epoch.